AmahiAnywhere/AmahiAnywhere/ConnectionTableViewController.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  ConnectionTableViewController.swift
+//  AmahiAnywhere
+//
+//  Created by Syed on 21/02/18.
+//  Copyright © 2018 Amahi. All rights reserved.
+//
+
+import UIKit
+
+class ConnectionTableViewController: UITableViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
+
+        // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
+        // self.navigationItem.rightBarButtonItem = self.editButtonItem
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        tableView.backgroundColor = UIColor.black
+    }
+
+    // MARK: - Table view data source
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        print(indexPath.row)
+        tableView.deselectRow(at: indexPath, animated: true)
+        if tableView.cellForRow(at: indexPath)? .accessoryType == .checkmark {
```

Comment:
1. Instead of equating an optional i.e. `tableView.cellForRow(at: indexPath)?. accessoryType` to .checkmark. you might have defined a cell if it existed.

Also, asking the view/ cell parameter is a not a good idea. You should not be dependent upon the current accessoryType. Instead, you should be having a data source which helps u assign the appropriate accessoryType.